content/working_with/manager/using-secrets.md

```diff
@@ -8,27 +8,25 @@ aliases: /manager/using-secrets/
-it's a symmetric encryption method which makes sure that the message encrypted cannot be manipulated/read without the key.
-When you create a secret, the key value can be a text string or it can be a file that contains the key value. With the secrets store, you can make sure all secrets (for example credentials to IaaS environments) are stored separately from blueprints, and adhere to isolation requirements between different tenants. You can include the secret's key in your blueprints and not include the actual values in the blueprints.
+The values of the secrets are encrypted in the database.
+We use the Fernet encryption of cryptography library, which is a symmetric encryption method that makes sure that the message encrypted cannot be manipulated/read without the key.
+When you create a secret, the key value can be a text string or it can be a file that contains the key value. The secret store lets you make sure all secrets (for example credentials to IaaS environments) are stored separately from blueprints, and that the secrets adhere to isolation requirements between different tenants. You can include the secret key in your blueprints and not include the actual values in the blueprints.
 For more information, see the [get_secret]({{< relref "developer/blueprints/spec-intrinsic-functions.md#get-secret" >}}) intrinsic function.
 
-### Hidden value secret
+### Secrets with a hidden value
 
-All the secrets values are encrypted in the DB. When you create a secret you can specify if you want its value to be hidden or not.
-A hidden-value secret means the secret's value will only be shown to the user who created it, tenant managers and sys-admins.
-Use of the secret is allowed according to the users roles and the visibility of the secret.
+All values for secrets are encrypted in the database. When you create a secret you can specify if you want its value to be hidden or not.
+A secret with a hidden value means the value is only shown to the user who created it, tenant managers and sys-admins.
+Users can use the secret according to the user roles and the visibility of the secret.
 
-#### Updating a hidden-value secret
+#### Updating a secret with a hidden value
 
-A user is hidden-value permitted if he is the creator of the secret, or a sys-admin or a tenant manager in the secret's tenant.
-Updating and deleting a hidden-value secret is allowed only to hidden-value permitted users.
-Updating a non-hidden value secret is allowed according to the users roles and the visibility of the secret.
-However, only hidden-value permitted users can update non-hidden value secret to be hidden-value.
+Only the creator of the secret, or a sys-admin or a tenant manager of the tenant with the secret can make, see, update or delete the hidden value.
+Users can change a hidden value into a shown value according to user roles and the visibility of the secret.
```

Comment:
I know I wrote it to complicated but I'm not sure the point is clear.
That users in the secret's tenant can update the secret but can't change the secret to be hidden value, only if the user is the creator of the secret, or a sys-admin or a tenant manager in the secret's tenant.